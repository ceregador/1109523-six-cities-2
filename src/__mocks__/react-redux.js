export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
