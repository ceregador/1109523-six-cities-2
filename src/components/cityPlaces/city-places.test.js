import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import Renderer from 'react-test-renderer';
import CityPlaces from './city-places.jsx';
import RentObjectCardList from '../rentObjectCardList/rent-object-card-list.jsx';
import OffersMap from '../offersMap/offers-map.jsx';

jest.mock(`../rentObjectCardList/rent-object-card-list.jsx`);
jest.mock(`../offersMap/offers-map.jsx`);
jest.mock(`react-redux`);
jest.mock(`../../selectors/active-city-selector`);

it(`renders correctly`, () => {
  const tree = Renderer
    .create(
        <CityPlaces
          activeCity={{name: `City`}}
          updateActiveCard={() => null}
        />)
    .toJSON();

  expect(RentObjectCardList).toHaveBeenCalled();
  expect(OffersMap).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({});
  expect(mappedProps).toHaveProperty(`activeCity`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`updateActiveCard`);

  expect(connectAdvanced).toHaveBeenCalledWith(CityPlaces);
});
