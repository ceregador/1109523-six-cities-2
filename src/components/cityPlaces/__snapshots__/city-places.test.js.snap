// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cities__places-container container"
>
  <div
    className="cities__right-section"
  />
</div>
`;
